flink-end-to-end-tests/test-scripts/test_yarn_kerberos_docker.sh

```diff
@@ -92,34 +88,28 @@ do
     sleep 2
 done
 
-CLUSTER_STARTED=1
-for (( i = 0; i < $CLUSTER_SETUP_RETRIES; i++ ))
-do
-    if start_hadoop_cluster; then
-       echo "Cluster started successfully."
-       CLUSTER_STARTED=0
-       break #continue test, cluster set up succeeded
-    fi
-
-    echo "ERROR: Could not start hadoop cluster. Retrying..."
-    docker-compose -f $END_TO_END_DIR/test-scripts/docker-hadoop-secure-cluster/docker-compose.yml down
-done
-
-if [[ ${CLUSTER_STARTED} -ne 0 ]]; then
+if ! retry_times $CLUSTER_SETUP_RETRIES 0 start_hadoop_cluster; then
     echo "ERROR: Could not start hadoop cluster. Aborting..."
     exit 1
 fi
 
+mkdir -p $FLINK_TARBALL_DIR
+tar czf $FLINK_TARBALL_DIR/$FLINK_TARBALL -C $(dirname $FLINK_DIR) .
```

Comment:
They are moved here to be run later. So if the test fails early (for example on cluster spin up), it won't create unused tarball.
This change was part of the test refactoring (you can check it by reading commits one-by-one).

Reply:
Yes, I was reading commits one by one and I was still confused (hence my question). I wasn't (and I'm still not) sure if all of those refactorings in the first commit are related to one another or if they are 3 independent things - if the latter one, it would be better to have them in separate commits.

Reply:
> I wasn't (and I'm still not) sure if all of those refactorings in the first commit are related to one another or if they are 3 independent things 

Bumping the question :)